### PR TITLE
Add !saveoutfit, !removeoutfit and extra flexibility to !avataritem

### PR DESCRIPTION
**Pull request by cocooo23**
This update adds some features for donors that make using commands much easier. Firstly, !saveoutfit and !removeoutfit have been added, as well as :saveoutfit and :removeoutfit which moderators can use on other people. !saveoutfit saves your current humanoid description and reapplies it every time you respawn. This way, if you were to give yourself a dominus via !avataritem, you wouldn't need to rerun that command every single time you respawned (saves a ton of time!!). 

The reversal of this command is !removeoutfit. It's used to cancel !saveoutfit reapplying your appearance and to also revert your character's appearance easily without multiple commands or respawning. Lastly, !avataritem has been updated to allow for custom hats. This is a feature inspired by another very popular admin system and some nice in-game avatar editors that allow this behavior. Users are able to give themselves external hats which the game creator can make or take from the catalog. This can be very useful for roleplay games or for devs with less scripting knowledge. The command has robust security and has a list of whitelisted classes which are the only ones allowed to be present in the accessory.

https://github.com/Epix-Incorporated/Adonis/assets/125225881/d4412186-1577-41b9-8b1e-a7e3a77e1038

### CODE REVIEW

File: MainModule/Server/Commands/Moderators.lua [hunk 1]
```diff
@@ -5261,6 +5261,76 @@ return function(Vargs, env)
...
+			AdminLevel = "Moderators";
+			Function = function(plr: Player, args: {[number]:string})
+				for _, v in service.GetPlayers(plr, args[1]) do
+					local connection = service.OutfitCache[v]
+					if connection then
+						connection:Disconnect()
+					end
+
+					if v.Character then
+						local humanoid: Humanoid? = v.Character and v.Character:FindFirstChildOfClass("Humanoid")
```
Comment (watameln) [inline MainModule/Server/Commands/Moderators.lua:5278]: what the devils

Comment (cocooo23) [reply MainModule/Server/Commands/Moderators.lua:5278]: i mess somethin up?

**Comment by cocooo23** [comment]
okie

**Comment by LiveFireExercise** [comment]
Hmm, fascinating

**Comment by cocooo23** [comment]
ello

**Comment by cocooo23** [comment]
any updates on here?

**Comment by AlexanderDarmody** [comment]
interesting

**Comment by cocooo23** [comment]
> interesting

truly

**Comment by LiveFireExercise** [comment]
> any updates on here?

yea im deciding whether to merge it or not

**Pull request merged**